src/transaction_builder.js

```diff
@@ -339,6 +338,10 @@ function prepareInput (input, kpPubKey, redeemScript, witnessValue, witnessScrip
     signScript = prevOutScript
   }
 
+  if (witness && !types.Satoshi(witnessValue)) {
+    throw new Error('Input was witness but not given witness value')
```

Comment:
ACK

Reply:
WitnessValue won't be mandatory if Transaction.addInput provides the value, right? (In some use cases the signer has no informations about the unspent value he's signing)

Reply:
Transaction.addInput probably shouldn't know the input value, since serializing would discard it. That information must _always_ be known by the signer (or even for extracting signatures) since otherwise they would fail to produce the sigHash. 

Requiring the scriptpubkey/value is a mission for signing operation is a mission for me atm, so the signer should always have this information

Reply:
@afk11 it is worth capturing that information though.

Reply:
See https://github.com/bitcoinjs/bitcoinjs-lib/pull/882

Reply:
got it